docs/lesson3.md

```diff
@@ -0,0 +1,120 @@
+
+# Hands-On Lab: Testing with GitHub Copilot
+
+## Overview
+
+**Goal:**  
+By the end of this session, you will learn basics of how GitHub Copilot can help with generating and executing test cases.
+
+**Estimated Duration:**  
+[Time in minutes/hours]
+
+**Audience:**  
+[developers, ops, testers, etc.]
+
+**Prerequisites:**  
+- Completed [Lesson 1: Installing and Configuring GitHub Copilot](docs/lesson1.md), or
+- Access to GitHub Copilot (subscription),
+- Visual Studio Code,
+- GitHub Copilot extension to VS Code.
+---
+
+## Table of Contents
+1. [Lab Objectives](#lab-objectives)
+2. [Environment Setup](#environment-setup)
+3. [Walkthrough](#walkthrough)
+    - [Step 1: Generating Unit Tests](#step-1-generating-unit-tests)
+    - [Step 2: Generating Integration Tests](#step-2-generating-integration-tests)
+4. [Troubleshooting](#troubleshooting)
+5. [Conclusion](#conclusion)
+
+---
+
+## Lab Objectives
+  
+- Understand how to leverage GitHub Copilot to generate and execute unit and integration tests.  
+- Learn how incremental test improvements can be achieved.  
+
+---
+
+## Environment Setup
+
+### Tools and Resources
+- GitHub Copilot enabled on your GitHub account.
+- Visual Studio Code with GitHub Copilot extension installed.
+- Code repository for the lab: [GitHub Copilot](https://github.com/neudesic/learning-github-copilot).
+
+### Steps to Prepare
+1. Clone the repository:  
+   ```bash
+   git clone https://github.com/neudesic/learning-github-copilot.git
+   cd learning-github-copilot/samples/eShop
+   ```
+2. Open the project in your IDE:  
+   ```bash
+   code .
+   ```
+
+---
+
+## Walkthrough
+
+### Step 1: Generating Unit Tests
+**Description:**  
+Provide a short description of the task or feature being demonstrated.  
+
+**Instructions:**  
+1. In VS Code, open file **src/Basket.API/Model/BasketItem.cs**
+2. Open Chat View by pressing Ctrl-Alt-I.
+3. Enter following request "generate comprehensive suite of unit test for this class. Include edge cases." and press Enter.  In a few moments, a code for new class *BasketItemTests* will show in the Chat View. 
+4. Copy the generated code and save it in a new file named *BasketItemTests.cs* in the **tests/Basket.UnitTests/** folder.
+5. Inspect the code in the file. Notice that GitHub Copilot recognized test frameworks used in the solution.
+6. In the Chat View prompt, enter "how do I run these tests?".
+7. The response will include instructions to open the terminal in the test project folder and execute tests. Follow the instructions or executed the following (assuming current directory is **/eShop/):
+   ```bash
+   cd tests/Basket.UnitTests
+   dotnet test
+   ```
+
+**Expected Outcome:**  
+The existing and new unit tests should pass. If some test cases fail, you can use GitHub Copilot to fix the code or the failing unit tests.
```

Comment:
Can you elaborate on how to fix the code?

Reply:
I can't predict the code or tests generated, so general advice there.

Reply:
I understand that. instead of saying fixing mention something around the lines of a additional prompting.